File: BST/bst.c
#include <stdio.h>
#include <stdlib.h>

struct node{
	int key;
	struct node *left,*right;
};

struct node* createnode(int value){
	struct node *temp = (struct node*)malloc(sizeof(struct node));
	temp->key = value;
	temp->left = temp->right = NULL;
	return temp;
}
struct node* insert(struct node* node, int value){
	if(node == NULL)
		return createnode(value);
	if(value < node->key)
		node->left = insert(node->left, value);
	else if(value > node->key)
		node->right = insert(node->right, value);
	return node;
}
void inorder(struct node* node){
	if(node != NULL){
		inorder(node->left);
		printf("%d  \n",node->key);
		inorder(node->right);
	}
}
int main(void) {
	struct node *root=NULL;
	root = insert(root, 100);
	insert(root, 90);
	insert(root, 110);
	insert(root, 80);
	insert(root, 95);

	inorder(root);
	return 0;
}

File: BST/bst.py
class Node:
	def __init__(self, key):
		self.left = None
		self.right = None
		self.value = key

def insert(root,node):
	if root is None:
		root = node

	else:
		if root.value < node.value:
			if root.right is None:
				root.right = node
			else:
				insert(root.right, node)
		else:
			if root.left is None:
				root.left = node
			else:
				insert(root.left, node)

def inorder(root):
	if root:
		inorder(root.left)
		print(root.value)
		inorder(root.right)

root = Node(100)
insert(root, Node(90))
insert(root, Node(110))
insert(root, Node(80))
insert(root, Node(95))

inorder(root)